com.avaloq.tools.ddk.xtext/src/com/avaloq/tools/ddk/xtext/resource/persistence/DirectLinkingResourceStorageLoadable.java

```diff
@@ -286,6 +295,15 @@ protected void loadEntries(final StorageAwareResource resource, final ZipInputSt
     }
   }
 
+  private void addFakeAST(final StorageAwareResource resource) {
+    InternalEObject[] values = new InternalEObject[] {new Container()};
+    BasicEList<InternalEObject> internalEObjectList = new BasicEList<InternalEObject>(); // NOPMD LooseCoupling
+    internalEObjectList.setData(1, values);
+    @SuppressWarnings("unchecked")
+    InternalEList<InternalEObject> internalEObjects = (InternalEList<InternalEObject>) (InternalEList<?>) resource.getContents();

```

Comment:
what about adding subclassed BasicEList which throws an UnsoportedOperationException in all operations? That way we are sure that we are not getting wrong results because we access a partially loaded resource.

Reply:
No, that's not going to work: this list is returned from Resource.getContents(), the inference container is contained in it